video/out/drm_atomic.c

```diff
@@ -458,3 +497,113 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
+            p->data.hdmi_metadata_type1.eotf = DRM_METADATA_EOTF_HLG;
+            break;
+        default:
+            // Default is SDR
+            p->data.hdmi_metadata_type1.eotf = DRM_METADATA_EOTF_TRADITIONAL_SDR;
+    }
+
+    if(p->data.hdmi_metadata_type1.eotf) {
+        // Convert to nits (cd/m2):
+        float min_luminance = 0.0;
+        float max_luminance = color->sig_peak * MP_REF_WHITE;
+        float max_cll = max_luminance;
+        float max_fall = max_luminance;
+        
+        // Range 0x0000 to 0xC350 maps to 0.0 - 1.0000. Step of 0.00002. Same for White Point
+        p->data.hdmi_metadata_type1.display_primaries[0].x = clamp16_round(colors.red.x);
+        p->data.hdmi_metadata_type1.display_primaries[0].y = clamp16_round(colors.red.y);
+        p->data.hdmi_metadata_type1.display_primaries[1].x = clamp16_round(colors.green.x);
+        p->data.hdmi_metadata_type1.display_primaries[1].y = clamp16_round(colors.green.y);
+        p->data.hdmi_metadata_type1.display_primaries[2].x = clamp16_round(colors.blue.x);
+        p->data.hdmi_metadata_type1.display_primaries[2].y = clamp16_round(colors.blue.y);
+
+        // White point
+        p->data.hdmi_metadata_type1.white_point.x = clamp16_round(colors.white.x);
+        p->data.hdmi_metadata_type1.white_point.y = clamp16_round(colors.white.y);
+
+        // Min Luminance: 16-bit value in units of 0.0001 cd/m2, where 0x0001 represents 0.0001 cd/m2 and 0xFFFF represents 6.5535 cd/m2.
+        p->data.hdmi_metadata_type1.min_display_mastering_luminance = clamp16(floorf(min_luminance * 10000.0));
+
+        // Max Luminance: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_display_mastering_luminance = clamp16(ceilf(max_luminance));
+
+        // Max Content Light Level: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_cll = clamp16(ceilf(max_cll));
+
+        // Max Frame Average Light Level: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_fall = clamp16(ceilf(max_fall));
+
+        drmModeCreatePropertyBlob(ctx->connector->fd, &p->data, sizeof p->data, &p->blob_id);
+        drmModeAtomicAddProperty(ctx->request, ctx->connector->id, property_hdr_id, p->blob_id);
+    }
+#endif
+}
+
+// switch HDR off
+void drm_destroy_hdrmeta(struct drm_atomic_context *ctx) {
+#if HAVE_DRM_HDR
+    if (ctx->hdr_metadata.blob_id) {
+        drmModeDestroyPropertyBlob(ctx->connector->fd, ctx->hdr_metadata.blob_id);
+        ctx->hdr_metadata.blob_id = 0;
+
+        drmModeAtomicReqPtr request = drmModeAtomicAlloc();
+        if (!request)
+             return;
+
+        // TODO: Is this necessary? https://github.com/mpv-player/mpv/pull/8456#issuecomment-778899198
+        // switch back to SDR
```

Comment:
calling `drmModeAtomicCommit` is currently within the responsibility of context_drm_egl.c, so IMO that's better kept there
plus this whole function can also be deleted if you called `drm_send_hdrmeta` with an SDR colorspace instead

Reply:
also if SDR is the fallback like you said, removing the HDR metadata should(?) be enough for a reset